node/src/components/consensus/consensus_protocol.rs

```diff
@@ -11,25 +11,42 @@ use crate::{
 
 /// Information about the context in which a new block is created.
 #[derive(Clone, DataSize, Eq, PartialEq, Debug, Ord, PartialOrd, Hash)]
-pub struct BlockContext {
+pub(crate) struct BlockContext<C>
+where
+    C: Context,
+{
     timestamp: Timestamp,
-    height: u64,
+    ancestor_values: Vec<C::ConsensusValue>,
```

Comment:
I'm reviewing the PR commit-by-commit, it would be nice to have the explanation (in the description of the commit) why we're adding the ancestor values.

Reply:
That would mean force-pushing, though. :grimacing: 

Concretely, I added `ancestor_values` to the block context because they make sense there: They are part of the context information that you need, both to propose a new block payload, and to validate an incoming block payload. It also allows us to pass fewer arguments in some places, where we had both context (or timestamp) and ancestor values.

Reply:
Oh, definitely not force-push. Just a general comment for the future.